### DISCUSSION TITLE
Help with Running Taiko in Firefox

### DISCUSSION THREAD

**Discussion by DevaIrakkamRubavathy** [getgauge/taiko#2160] [Development]
Hi! I'm trying to run npx taiko on MacOS (intel, if it matters) with a Firefox browser, it is throwing the following error when I am giving openBrowser() command :

```
TAIKO_BROWSER_PATH="/Applications/Firefox.app/Contents/MacOS/firefox" taiko

Version: 1.2.5 (/Applications/Firefox.app/Contents/MacOS/firefox)
Type .api for help and .exit to quit

> openBrowser()
Target.getTargetInfo (RemoteAgentError@chrome://remote/content/cdp/Error.jsm:29:5
Uncaught UnknownMethodError@chrome://remote/content/cdp/Error.jsm:112:7
execute@chrome://remote/content/cdp/domains/DomainCache.jsm:98:13
receiveMessage@chrome://remote/content/cdp/sessions/ContentProcessSession.jsm:84:45
)
```

If anybody know how to fix this, please help me out.
TIA!

**Comment by vitali2y** [comment]
Almost 2 years passed, and the same problem for me on _Linux Mint 21_ with the latest nightly _Firefox 115.0a1.en-US.linux-x86_64_:
```
➜  ~ ✗ TAIKO_BROWSER_PATH=~/.local/firefox/firefox npx taiko          

Version: 1.3.10 (/home/vit/.local/firefox/firefox)
Type .api for help and .exit to quit

> openBrowser()
Target.getTargetInfo (RemoteAgentError@chrome://remote/content/cdp/Error.sys.mjs:20:5
Uncaught UnknownMethodError@chrome://remote/content/cdp/Error.sys.mjs:103:7
execute@chrome://remote/content/cdp/domains/DomainCache.sys.mjs:92:13
receiveMessage@chrome://remote/content/cdp/sessions/ContentProcessSession.sys.mjs:79:45
)
>
```
Any ideas, tricks, etc?

**Comment by xiaorun2969** [comment]
A possible useful attempt is to add parameters when opening a browser, such as:  openBrowser({args:["--no-sandbox","--disable-gpu","--disable-dev-shm-usage","--disable-setuid-sandbox","--no-zygote"]})

**Comment by snowman** [comment]
If you want to get the current tab url of Firefox (which `Target.getTargetInfo()` does NOT work), check [How to check the current Chrome tab url? #449 - cyrus-and/chrome-remote-interface](https://github.com/cyrus-and/chrome-remote-interface/issues/449#issuecomment-2028017588)
